### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by supermamie** [comment]
Upgrade done on my server, successfully.
(And no, my server is not on bookworm yet)

**Comment by supermamie** [comment]
I just tested the upgrade with all those changes, and it failed : 
https://paste.yunohost.org/raw/otaqufequp

Here are the last lines of the log : 
```
2024-04-03 11:05:39,300: DEBUG - + apt-get --assume-yes --quiet -o=Acquire::Retries=3 -o=Dpkg::Use-Pty=0 --no-remove --option Dpkg::Options::=--force-confdef --option Dpkg::Options::=--force-confold install libcurl4-gnutls-dev --dry-run
2024-04-03 11:05:40,486: WARNING - The following packages have unmet dependencies:
2024-04-03 11:05:40,487: WARNING -  libcurl4-gnutls-dev : Conflicts: libcurl4-openssl-dev but 7.74.0-1.3+deb11u11 is to be installed
2024-04-03 11:05:40,487: WARNING -  libcurl4-openssl-dev : Conflicts: libcurl4-gnutls-dev but 7.74.0-1.3+deb11u11 is to be installed
2024-04-03 11:05:40,488: DEBUG - + ynh_die '--message=Unable to install dependencies'
2024-04-03 11:05:40,538: DEBUG - + ynh_exit_properly
2024-04-03 11:05:40,538: DEBUG - + [[ provision_or_update_apt =~ ^install$|^upgrade$|^restore$ ]]
2024-04-03 11:05:40,541: WARNING - Unable to install dependencies
2024-04-03 11:05:42,137: ERROR - provision_or_update failed for apt : An error occured inside the script snippet
```

**Comment by alexAubin** [comment]
@supermamie : hmmmyes that's an annoying conflict between `libcurl4-openssl-dev` and `libcurl4-gnutls-dev` ... Not sure how to deal with that, my guess is that for some reason you already have `libcurl4-openssl-dev` or `libcurl4-gnutls-dev` installed ...

Can you check this:

```
apt install aptitude
aptitude why libcurl4-openssl-dev
aptitude why libcurl4-gnutls-dev
```

**Comment by alexAubin** [comment]
!testme

File: manifest.toml [hunk 1]
```diff
@@ -113,6 +109,21 @@ ram.runtime = "50M"
...
 
     [resources.permissions]
+    main.allowed = "visitors"
 
     [resources.apt]
-    packages = "build-essential, libirrlicht-dev, cmake, libbz2-dev, libpng-dev, libjpeg-dev, libxxf86vm-dev, libgl1-mesa-dev, libsqlite3-dev, libogg-dev, libvorbis-dev, libopenal-dev, libcurl4-dev, libfreetype6-dev, zlib1g-dev, libgmp-dev, libjsoncpp-dev, libluajit-5.1-dev, libncurses-dev, libzstd-dev"
+    packages = ["build-essential", 
+                "cmake", 
+                "libsqlite3-dev",
+                "libcurl4-gnutls-dev",
```
Comment (alexAubin) [inline manifest.toml:118]: Hmkay so i realize this dependency was added to the list though i'm not sure why x_X, naively I'd say we can remove it

**Comment by supermamie** [comment]
> ```
> aptitude why libcurl4-openssl-dev
> aptitude why libcurl4-gnutls-dev
> ```

Here are the results : 
```
root@myserver:~# aptitude why libcurl4-openssl-dev
i   minetest-ynh-deps    Dépend  libcurl4-dev
i A libcurl4-openssl-dev Fournit libcurl4-dev
root@myserver:~# aptitude why libcurl4-gnutls-dev
i   python3-reportbug    Dépend  python3-debianbts (>= 2.10)
i A python3-debianbts    Dépend  python3-pysimplesoap       
i A python3-pysimplesoap Dépend  python3-pycurl             
i A python3-pycurl       Suggère libcurl4-gnutls-dev   
```

File: manifest.toml (hunk 1 above)
Comment (orhtej2) [reply manifest.toml:118]: I've changed it to flavour explicitly mentioned by [the docs](https://github.com/minetest/minetest/blob/stable-5/doc/compiling/linux.md) as the virtual package `libcurl4-dev` did not resolve properly no `bookworm` [here](https://ci-apps-bookworm-dev.yunohost.org/ci/job/289). We can try reverting as the problem boiled down to `luajit` version in the end 🤷

**Comment by orhtej2** [comment]
Checking again with generic `libcurl4-dev`....: https://github.com/YunoHost-Apps/minetest_ynh/pull/61

**Comment by orhtej2** [comment]
Seems to be working, can you @supermamie try **after creating a proper backup**?

```
sudo yunohost app upgrade minetest -u https://github.com/orhtej2/minetest_ynh/tree/patch-1
```

**Comment by supermamie** [comment]
@orhtej2 your branch is working on my server :+1:

**Comment by tituspijean** [comment]
!testme
!bookwormtestme

**Comment by tituspijean** [comment]
Here upgrade from 7502e815 is failing due to a typo in the old manifest.json which was not fixed before migrating to packaging v2. Let's remove that test when we release YunoHost 12?

**Comment by alexAubin** [comment]
(Yes, level 8->6 regressions like this are expected when for major debian upgrades, it's not a huge deal)

**Review by alexAubin — APPROVED** [review]

**Pull request merged**